### PR TITLE
LPS-92741 analyticsKey is not used anymore

### PR DESCRIPTION
**Pull request by marcellustavares**

### CODE REVIEW

File: modules/apps/analytics/analytics-client-js/src/analytics.js [hunk 1]
```diff
@@ -159,22 +160,28 @@ class Analytics {
...
+		if (!storedUserId || ((storedAnonymous === false) && (anonymous === true))) {
+			const userId = this._generateUserId();
 
+			this._persist(STORAGE_KEY_ANONYMOUS, anonymous);
 			this._persist(STORAGE_KEY_USER_ID, userId);
 
 			return Promise.resolve(userId);
 		}
+		else {
+			return Promise.resolve(userId);
```
Comment (andreldm) [inline modules/apps/analytics/analytics-client-js/src/analytics.js]: Is this right? I mean, the userId seems to not exist in the else scope

Comment (marcellustavares) [reply modules/apps/analytics/analytics-client-js/src/analytics.js]: You're right, it should be `storedUserId`, I forgot to update this variable.

Comment (marcellustavares) [reply modules/apps/analytics/analytics-client-js/src/analytics.js]: Fixed, 9e9ad934b1a2e603cc42e554ee0aed20967e558c

File: modules/apps/analytics/analytics-client-js/src/analytics.js [hunk 2]
```diff
@@ -5,7 +5,6 @@ import middlewares from './middlewares/defaults';
 import Client from './Client';
 
 import defaultPlugins from './plugins/defaults';
-import fingerprint from './utils/fingerprint';
```
Comment (andreldm) [inline modules/apps/analytics/analytics-client-js/src/analytics.js]: Is it possible to get rid of `fingerprint.js`?

File: modules/apps/analytics/analytics-client-js/src/utils/hash.js [hunk 3]
```diff
@@ -1,11 +1,8 @@
 import objectHash from 'object-hash';
 
-const ignoredProps = ['screenWidth', 'screenHeight', 'devicePixelRatio'];
-
 const hash = value =>
 	objectHash(value, {
 		algorithm: 'md5',
```
Comment (andreldm) [inline modules/apps/analytics/analytics-client-js/src/utils/hash.js]: md5 is too weak, any reason not to use SHA1 or SHA256?

File: modules/apps/analytics/analytics-client-js/src/analytics.js [hunk 4]
```diff
@@ -193,15 +193,9 @@ class Analytics {
 		};
 
 		const storedIdentityHash = storage.get(STORAGE_KEY_IDENTITY_HASH);
-		let newIdentityHash = hash(bodyData);
+		const newIdentityHash = hash(bodyData);
```
Comment (andreldm) [inline modules/apps/analytics/analytics-client-js/src/analytics.js]: Now that the browser fingerprint is not used, does this hash become more likely to collide?

File: modules/apps/analytics/analytics-client-js/src/analytics.js (hunk 2 above)
Comment (interaminense) [reply modules/apps/analytics/analytics-client-js/src/analytics.js]: I think so, it won't be necessary anymore.

**Comment by interaminense** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by interaminense** [comment]
Just started reviewing :)

**Pull request closed without merging**